src/structlog/tracebacks.py

```diff
@@ -0,0 +1,261 @@
+
+    return obj_repr
+
+
+def extract(
+    exc_type: Type[BaseException],
+    exc_value: BaseException,
+    traceback: Optional[TracebackType],
+    *,
+    show_locals: bool = False,
+    locals_max_string: int = LOCALS_MAX_STRING,
+) -> Trace:
+    """
+    Extract traceback information.
+
+    :param exc_type: Exception type.
+    :param exc_value: Exception value.
+    :param traceback: Python Traceback object.
+    :param show_locals: Enable display of local variables. Defaults to False.
+    :param locals_max_string: Maximum length of string before truncating, or
+        ``None`` to disable.
+    :param max_frames: Maximum number of frames in each stack
+
+    :returns: A Trace instance with structured information about all
+        exceptions.
+
+    .. versionadded:: 22.1
+    """
+
+    stacks: List[Stack] = []
+    is_cause = False
+
+    while True:
+        stack = Stack(
+            exc_type=safe_str(exc_type.__name__),
+            exc_value=safe_str(exc_value),
+            is_cause=is_cause,
+        )
+
+        if isinstance(exc_value, SyntaxError):
+            stack.syntax_error = SyntaxError_(
+                offset=exc_value.offset or 0,
+                filename=exc_value.filename or "?",
+                lineno=exc_value.lineno or 0,
+                line=exc_value.text or "",
+                msg=exc_value.msg,
+            )
+
+        stacks.append(stack)
+        append = stack.frames.append  # pylint: disable=no-member
+
+        for frame_summary, line_no in walk_tb(traceback):
+            filename = frame_summary.f_code.co_filename
+            if filename and not filename.startswith("<"):
+                filename = os.path.abspath(filename)
+            frame = Frame(
+                filename=filename or "?",
+                lineno=line_no,
+                name=frame_summary.f_code.co_name,
+                locals={
+                    key: to_repr(value, max_string=locals_max_string)
+                    for key, value in frame_summary.f_locals.items()
+                }
+                if show_locals
+                else None,
+            )
+            append(frame)
+
+        cause = getattr(exc_value, "__cause__", None)
+        if cause and cause.__traceback__:
+            exc_type = cause.__class__
+            exc_value = cause
+            traceback = cause.__traceback__
+            is_cause = True
+            continue
+
+        cause = exc_value.__context__
+        if (
+            cause
+            and cause.__traceback__
+            and not getattr(exc_value, "__suppress_context__", False)
+        ):
+            exc_type = cause.__class__
+            exc_value = cause
+            traceback = cause.__traceback__
+            is_cause = False
+            continue
+
+        # No cover, code is reached but coverage doesn't recognize it.
+        break  # pragma: no cover
+
+    trace = Trace(stacks=stacks)
+    return trace
+
+
+class JSONFormatter:
```

Comment:
So it just occurred to me that this formatter (and everything related to it) shouldn't carry the JSON name, because it's misleading.

While in 99.9% of cases the structured exceptions are gonna be serialized into JSON, it doesn't format it to JSON!

I realize it's gonna be somewhat less-obvious to document, but OTOH we'd be underselling it, if we'd call it this way.

Maybe `StructuredFormatter`? I'm open to better names, but JSON is shouldn't be it.

Reply:
Screw you! I was thinking the same and either had no better name.  So I hoped I could get away with `JSONFormatter` 🤪

I’ll try to hit Ballmer's Peek tonight and see if I can come up with a better name.  😬

Reply:
Maybe just `DictFormatter` since it returns a list of trace dicts ...

Reply:
I was gonna suggest DictFormatter but was detracted by the fact that it returns a list...but I guess it works anyways and is nicer to type/say.

Reply:
What about `json_tracebacks`?  Whould we rename it to `dict_tracebacks` accordingly? But OTOH it looks quite good as it is in its usage context:
```python
         ...     processors=[
         ...         structlog.processors.json_tracebacks,
         ...         structlog.processors.JSONRenderer(),
         ...     ],
```

Reply:
It looks good, but it's still wrong. :)

Reply:
done